### ISSUE TITLE
pre-commit hook args ignore rules and other options passed in

### ISSUE THREAD

**Issue by netopsengineer** [aristanetworks/j2lint#60]
Hello,

I can't seem to get the pre-commit hook to respect args passed in, and as such utilize any rules, custom, or built in.

```yaml
---
repos:
-   repo: https://github.com/aristanetworks/j2lint.git
    rev: v1.0.0
    hooks:
      - id: j2lint
        args: [--ignore 'jinja-statements-indentation']
```

Results in:

```shell
(poetry-3.10)  > pre-commit run --all-files
Check for Linting error on Jinja2 templates..............................Failed
- hook id: j2lint
- exit code: 2

JINJA2 LINT ERRORS
************ File nokia_sros/log_configuration.j2
nokia_sros/log_configuration.j2:6 Bad Indentation, expected 5, got 1 (jinja-statements-indentation)
nokia_sros/log_configuration.j2:7 Bad Indentation, expected 9, got 1 (jinja-statements-indentation)
nokia_sros/log_configuration.j2:9 Bad Indentation, expected 13, got 1 (jinja-statements-indentation)
nokia_sros/log_configuration.j2:11 Bad Indentation, expected 13, got 1 (jinja-statements-indentation)
```

**Comment by gmuloc** [comment]
Hi @netopsengineer, thanks for opening the issue and using j2lint.

I troubleshooted a bit around and this syntax would work:

```
    - repo: https://github.com/aristanetworks/j2lint.git
      rev: "v1.0.0"
      hooks:
        - id: j2lint
          name: Check for Linting errors on Jinja2 files
          args: [--ignore, jinja-statements-indentation, --]
```

Notice two things:
1. the fact that I am separating the `--ignore` and the `jinja-statements-indentation`
Indeed printing the received options from pre-commit with your syntax I get:
```
*** Received sys.args:
['--ignore jinja-statements-indentation', '--', 'ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2']
*** Parsed option
Namespace(files=['--ignore  jinja-statements-single-space', 'ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2'], ignore=[], warn=[], list=False, rules_dir=['/tmp/.cache/pre-commit/repo9wsz_uft/py_env-python3.10/lib/python3.10/site-packages/j2lint/rules'], verbose=False, debug=False, json=False, stdin=False, log=False, version=False, stdout=False)
```
and with my syntax:
```
*** Received sys.args:
['--ignore', 'jinja-statements-indentation', '--', 'ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2']
*** Parsed option
Namespace(files=['ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos-intended-config.j2'], ignore=['jinja-statements-indentation'], warn=[], list=False, rules_dir=['/tmp/.cache/pre-commit/repo9wsz_uft/py_env-python3.10/lib/python3.10/site-packages/j2lint/rules'], verbose=False, debug=False, json=False, stdin=False, log=False, version=False, stdout=False)
```


2. the ending `--` - cf https://github.com/aristanetworks/j2lint/issues/40 for the "why"



To ignore two rules:
```
    - repo: https://github.com/aristanetworks/j2lint.git
      rev: "v1.0.0"
      hooks:
        - id: j2lint
          name: Check for Linting errors on Jinja2 files
          args: [--ignore, S3, jinja-statements-single-space, --]
```


I see two things here that could me made better
* I am considering changing/refactoring the args declaration for ignore and warns flags as they impose ordering of arguments which is quite not convenient as described in #40 
* In the meantime I will add some comment regarding the proper syntax

I hope you can move forward with using j2lint with pre-commit for now with the info above. Let me know your thoughts!

**Comment by netopsengineer** [comment]
Hey @gmuloc thanks for the quick reply! j2lint is invaluable, I author alot of templates for network devices, and one forgotten closing tag can ruin your day.  The `pre-commit` journey is new, and to help onboard new NetDevOps Engineers into authoring and maintaining templates that generate golden configs, so catching things early is a big win, thanks for the time you guys put into the tool!

Great find on the weirdness around the arg parsing, and the `--` getting around the issue for the time being.  I can confirm that this is working well now, I swapped from my local custom hook that specified all the args in the `entry` to the one sourced from this repo, and all is well!
